commit bbbbf0639efb8987c293bb9209df148a7e556c0c
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Thu May 5 21:38:04 2022 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Thu May 5 21:38:04 2022 -0400

    To support Tensor used in the model weights for cross_validation. (#486)
    
    * To support Tensor used in the model weights for cross_validation.
    
    * Reemoved no used import.
---
 nvflare/private/fed/server/server_engine.py | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 326c4c67..b305816e 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -67,6 +67,18 @@ from .server_engine_internal_spec import EngineInfo, RunInfo, ServerEngineIntern
 from .server_status import ServerStatus
 
 
+class ClientConnection:
+    def __init__(self, client):
+        self.client = client
+
+    def send(self, data):
+        data = pickle.dumps(data)
+        self.client.send(data)
+
+    def recv(self):
+        return self.client.recv()
+
+
 class ServerEngine(ServerEngineInternalSpec):
     def __init__(self, server, args, client_manager: ClientManager, snapshot_persistor, workers=3):
         """Server engine.
@@ -586,6 +598,7 @@ class ServerEngine(ServerEngineInternalSpec):
             try:
                 address = ("localhost", port)
                 command_conn = CommandClient(address, authkey="client process secret password".encode())
+                command_conn = ClientConnection(command_conn)
 
                 self.run_processes[run_number][RunProcessKey.CONNECTION] = command_conn
             except Exception:
